plasmapy/formulary/parameters.py

```diff
@@ -663,23 +689,27 @@ def thermal_speed(
     >>> thermal_speed(1e6*u.K, "e-", method="mean_magnitude")
     <Quantity 621251... m / s>
     """
-    m = mass if np.isfinite(mass) else particles.particle_mass(particle)
+    if mass is None:
+        mass = particles.particle_mass(particle)
 
     # different methods, as per https://en.wikipedia.org/wiki/Thermal_velocity
     try:
-        coef = _coefficients[ndim]
+        coeff = thermal_speed_coefficients[ndim]
     except KeyError:
         raise ValueError("{ndim} is not a supported value for ndim in thermal_speed")
     try:
-        coef = coef[method]
+        coeff = coeff[method]
     except KeyError:
         raise ValueError("Method {method} not supported in thermal_speed")
 
-    return np.sqrt(coef * k_B * T / m)
+    speed = thermal_speed_lite(T=T.value, mass=mass.value, coeff=coeff)
+    return speed * u.m / u.s
```

Comment:
I know this is supposed to be the slow version, but on my machine the following is still twice as fast - not reason not to take that:
```suggestion
    return u.Quantity(speed, u.m / u.s)
```

(with a predefined speed unit `mps = u.m / u.s` there's another factor of 2 speed up)